### DISCUSSION TITLE
standardizing "beats" and allowing for variance

### DISCUSSION THREAD

**Discussion by iznaut** [NEW-CYLANDIA/little-warioware#17] [General]
i'm not exactly clear on the "standard" length of a microgame as they appear in WW - i tend to trust the very detailed Super Mario Wiki (excerpt below), but also 4 seconds feels _very_ short to me (i always thought it was 6 for some reason?)

i think there's some wiggle room with how the timer is displayed - sometimes it doesn't appear right away (is this only for IQ games?), so the game is technically longer, but the timer is always the same length.

some research likely needs to be done here to pin this down. also, to what extent do we allow submissions to tweak this? (may be something to think more about in the long-term rather than something that needs to be available right away)

>The unit of time for all microgames is beats. A standard microgame is 8 beats long (4 seconds, at normal speed), while microgames in the IQ genre, hosted by Orbulon, last 16 beats (8 seconds); the latter length is also used for the occasional microgame here-and-there in later games. Fronk microgames are only 4 beats (2 seconds) long and the WarioWatch has a timer for the whole stage rather than each individual microgame. In most games, the BPM starts out relatively slow and then increases as the player completes microgames. The average length of a microgame has been suggested by Nintendo to be five seconds, when in reality it is only four.[1] As more microgames are played, the game will show a "Speed Up!" message which signifies the increase in the microgames' speed.

Source: https://www.mariowiki.com/Microgame

**Comment by TaylorAnderson** [comment]
do we wanna maybe decide on x seconds for a standard game and then allow for multiples of that?

**Comment by lazerwalker** [comment]
Curious if you have any additional thoughts about this.

My instinct is to copy Nintendo for now ('standard' game is 8 beats, which is 4 seconds at normal speed) and maybe allow double-length games? Might be worth manually playing around with what it feels like at 4s vs 5s vs 6s.

We can have a centralized beat manager emit a signal every beat (a la Waltz), so if games want to bop along to the beat like actual WW they have the hooks to do so.
